kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -426,8 +375,8 @@ The solution is to use rewrite operations in a downward pass through the project
                    | toStack ( Int , Local )
 
   // retains information about the value that was deconstructed by a projection
-  syntax Context ::= CtxField( Ty, VariantIdx, List, Int )
-                   | CtxIndex( Ty, List , Int ) // array index constant or has been read before
+  syntax Context ::= CtxField( MaybeTy, VariantIdx, List, Int )
+                   | CtxIndex( MaybeTy, List , Int ) // array index constant or has been read before
```

Comment:
This should change back to `Ty`. 
It is symptomatic of the overwritten `TypedValue` in `#setLocalValue` above. The `Ty` comes from the Local that we are deconstructing, but with the change above we are getting `MaybeTy` on locals.
As an invariant, the `TypedValue`s  and `NewLocal`s in `<locals>` always have a `Ty`. 

:warning:  (as I am writing this I realise that the `Moved` probably also needs `Ty` and `Mutability` fields since now we allow overwriting a `Moved` with a new value and have to retain the information)